bigquery_etl/cli/backfill.py

```diff
@@ -368,3 +373,136 @@ def process(ctx, qualified_table_name, sql_dir, project_id, dry_run):
+    try:
+        client.get_table(backfill_staging_qualified_table_name)
+    except NotFound:
+        click.echo(
+            f"""
+            Backfill staging table does not exists for {qualified_table_name}:
+            {backfill_staging_qualified_table_name}
+            """
+        )
+        sys.exit(1)
+
+    project, dataset, table = qualified_table_name_matching(qualified_table_name)
+
+    # clone production table
+    cloned_table_id = f"{table}_backup_{entry_to_complete.entry_date}".replace("-", "_")
+    cloned_table_full_name = f"{BACKFILL_DESTINATION_PROJECT}.{BACKFILL_DESTINATION_DATASET}.{cloned_table_id}"
+    _copy_table(qualified_table_name, cloned_table_full_name, client, True)
+
+    # copy backfill data to production data
+    start_date = entry_to_complete.start_date
+    end_date = entry_to_complete.end_date
+    dates = [start_date + timedelta(i) for i in range((end_date - start_date).days + 1)]
+
+    # replace partitions in production table that have been backfilled
+    for backfill_date in dates:
+        partition = backfill_date.strftime("%Y%m%d")
+        if backfill_date not in entry_to_complete.excluded_dates:
+            production_table = f"{qualified_table_name}${partition}"
+            backfill_table = f"{backfill_staging_qualified_table_name}${partition}"
+            _copy_table(backfill_table, production_table, client)
+
+    # delete backfill staging table
+    client.delete_table(backfill_staging_qualified_table_name)
+    click.echo(
+        f"Backfill staging table deleted: {backfill_staging_qualified_table_name}"
+    )
+
+    click.echo(
+        f"Completed backfill for {qualified_table_name} with entry date {entry_to_complete.entry_date}"
+    )
+
+
+def _copy_table(
+    source_table: str, destination_table: str, client, clone: bool = False
+) -> None:
+    """
+    Copy and overwrite table from source to destination table.
+
+    If clone is True, clone (previous) production data for backup before swapping stage data into production.
+    """
+    job_type_str = "copied"
+
+    if clone:
+        copy_config = bigquery.CopyJobConfig(
+            write_disposition=bigquery.WriteDisposition.WRITE_EMPTY,
+            operation_type=bigquery.job.copy_.OperationType.CLONE,
+            destination_expiration_time=(datetime.now() + timedelta(days=30)).strftime(
+                "%Y-%m-%dT%H:%M:%SZ"
+            ),
+        )
+        job_type_str = "cloned"
+    else:
+        copy_config = bigquery.CopyJobConfig(
+            write_disposition=bigquery.WriteDisposition.WRITE_TRUNCATE,
+            operation_type=bigquery.job.copy_.OperationType.COPY,
+        )
+
+    try:
+        client.copy_table(
+            source_table,
+            destination_table,
+            job_config=copy_config,
+        ).result()
+    except Exception as e:
+        print(f"Unable to clone table {source_table} with error: {e}")
+        sys.exit(1)
```

Comment:
What's the usual Exception here? Might be worth catching directly

Reply:
The exception has been expanded to NotFound and Conflict.